client/my-sites/site-settings/wrap-settings-form.jsx

```diff
@@ -0,0 +1,163 @@
+				! nextProps.isSavingSettings
+			) {
+				if ( nextProps.isSaveRequestSuccessful ) {
+					nextProps.successNotice( nextProps.translate( 'Settings saved!' ), { id: 'site-settings-save' } );
+					nextProps.clearDirtyFields();
+					nextProps.markSaved();
+				} else {
+					let text;
+					switch ( nextProps.saveRequestError.error ) {
+						case 'invalid_ip':
+							text = nextProps.translate( 'One of your IP Addresses was invalid. Please try again.' );
+							break;
+						default:
+							text = nextProps.translate( 'There was a problem saving your changes. Please try again.' );
+					}
+					nextProps.errorNotice( text, { id: 'site-settings-save' } );
+				}
+			}
+		}
+
+		// Some Utils
+		handleSubmitForm = event => {
+			if ( ! event.isDefaultPrevented() && event.nativeEvent ) {
+				event.preventDefault();
+			}
+
+			this.submitForm();
+			this.props.trackEvent( 'Clicked Save Settings Button' );
+		};
+
+		submitForm = () => {
+			const { fields, site } = this.props;
+			this.props.removeNotice( 'site-settings-save' );
+			this.props.saveSiteSettings( site.ID, fields );
+		}
+
+		handleRadio = event => {
+			const currentTargetName = event.currentTarget.name,
+				currentTargetValue = event.currentTarget.value;
+
+			this.props.updateFields( { [ currentTargetName ]: currentTargetValue } );
+		};
+
+		handleToggle = name => () => {
+			this.props.trackEvent( `Toggled ${ name }` );
+			this.props.updateFields( { [ name ]: ! this.props.fields[ name ] } );
+		};
+
+		onChangeField = field => event => {
+			const { updateFields } = this.props;
+			updateFields( {
+				[ field ]: event.target.value
+			} );
+		};
+
+		render() {
+			const utils = {
+				handleRadio: this.handleRadio,
+				handleSubmitForm: this.handleSubmitForm,
+				handleToggle: this.handleToggle,
+				onChangeField: this.onChangeField,
+				submitForm: this.submitForm,
+			};
+
+			return (
+				<div>
+					<QuerySiteSettings siteId={ this.props.siteId } />
+					<SettingsForm { ...this.props } { ...utils } />
+				</div>
+			);
+		}
+	}
+
+	const connectComponent = connect(
+		state => {
+			const siteId = getSelectedSiteId( state );
+			const isRequestingSettings = isRequestingSiteSettings( state, siteId );
+			const isSavingSettings = isSavingSiteSettings( state, siteId );
+			const isSaveRequestSuccessful = isSiteSettingsSaveSuccessful( state, siteId );
+			const settings = getSiteSettings( state, siteId );
+			return {
+				isRequestingSettings: isRequestingSettings && ! settings,
+				isSavingSettings,
+				isSaveRequestSuccessful,
+				settings,
+				siteId
+			};
+		},
+		dispatch => {
+			const boundActionCreators = bindActionCreators( {
+				errorNotice,
+				recordTracksEvent,
+				removeNotice,
+				saveSiteSettings,
+				successNotice,
+			}, dispatch );
+			const trackEvent = name => dispatch( recordGoogleEvent( 'Site Settings', name ) );
+			const trackEventOnce = memoize( name => dispatch( recordGoogleEvent( 'Site Settings', name ) ) );
```

Comment:
While `mapDispatchToProps` is not called so frequently, apart when parent props changes and while testing it behaves correctly, I think we should move this inside the wrapperComponent to avoid bugs